automap/src/comm_layer/pcp_pmp_common/linux_specific.rs

```diff
@@ -153,17 +197,35 @@ mod tests {
         168.63.129.16 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
         169.254.169.254 via 10.1.0.1 dev eth0 proto dhcp src 10.1.0.84 metric 100\n\
         172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown";
-        let reg_for_any_other_line = ip_route_regex();
+        let regex = ip_route_regex();
 
         let lines = route_n_output.split('\n').collect::<Vec<&str>>();
 
         lines.iter().for_each(|line| {
-            assert!(
-                reg_for_any_other_line.is_match(line),
-                "Lines: {:?} line: {}",
-                lines,
-                line
-            );
+            assert!(regex.is_match(line), "Lines: {:?} line: {}", lines, line);
         });
     }
+
+    #[test]
+    fn reg_for_ip_route_bad_ip() {
```

Comment:
A way to make this test easier to read (and extend in the future) might be to make your `route_n_output` consist of a vector of tuples, like this:
```
let route_n_output = vec![
    ("10.1.0.0/16 dev eth0 proto kernel scope link src 10.1.0.84 metric 100", true, ""),
    ("0.1.0 dev eth0 proto dhcp scope link src 10.1.0.84 metric 100", false, "IPv4 address has only three elements"),
    [...]
];
```
Then you can apply the regex to the first item in each pair and expect the result in the second item of each pair, and if the second item was false and the regex succeeded, you'd be able to print the third item in the failure message. It'd be easier to see what you were expecting from each line, and easier to make sure that an additional line was correctly added.

Also, you could combine this test with the one above and have both successes and failures in the same test, possibly arranged to make aspects of your requirements clearer.